src/components/molecules/Wallet/Details.tsx

```diff
@@ -9,9 +9,10 @@ import { formatCurrency } from '@coingecko/cryptoformat'
 import { useUserPreferences } from '../../../providers/UserPreferences'
 
 export default function Details(): ReactElement {
-  const { balance, connect, logout, web3Provider } = useOcean()
+  const { balance, config, connect, logout, web3Provider } = useOcean()
   const { locale } = useUserPreferences()
   const [providerInfo, setProviderInfo] = useState<IProviderInfo>()
+  const provider = window.web3.currentProvider
```

Comment:
we need to always use the initialized `web3` & `web3provider` from `useOcean()` from within components. Otherwise this code here will break app in browsers without MetaMask installed. Also, `window.web3` is deprecated, see the browser console:
<img width="1231" alt="Screen Shot 2021-03-15 at 16 23 51" src="https://user-images.githubusercontent.com/90316/111177806-dfc38780-85aa-11eb-963d-b9e4ab057fa4.png">